test/README.md

```diff
@@ -6,35 +6,81 @@ Thanks for writing tests! Here's a quick run-down on our current setup.
 
 Please familiarise yourself with these if you plan on contributing! :+1:
 
- - [enzyme](https://github.com/airbnb/enzyme)
- - [mocha](https://mochajs.org)
- - [chai](http://chaijs.com)
- - [jsdom](https://github.com/tmpvar/jsdom)
-
+- [mocha](https://github.com/mochajs/mocha)
+- [enzyme](https://github.com/airbnb/enzyme)
+- [jsdom](https://github.com/tmpvar/jsdom)
+- [karma](https://github.com/karma-runner/karma)
+- [docker](https://github.com/docker/docker)
+- [chai](https://github.com/chaijs/chai)
+- [sinon](https://github.com/sinonjs/sinon)
 
 ## Commands
 
-##### Run the core mocha unit/integration test suite.
+Material-UI uses a wide range of tests approach as each of them comes with a different
+tradeoff. It's mainly a completeness/speed tradeoff.
+
+### React API level
+
+First, we have the **unit test** suite.
+It's using [mocha](https://mochajs.org) and the *shallow* API of [enzyme](https://github.com/airbnb/enzyme).
+It's allowing testing the components in isolation.
+It's the faster approach. It's best suited for testing many combinations.
+
+Then, we have the **integration** tests.
+We are using the *mount* API of [enzyme](https://github.com/airbnb/enzyme).
+It's allowing testing the integration of different components with a virtual DOM.
+This virtual DOM is provided by [jsdom](https://github.com/tmpvar/jsdom).
+It's here to make sure components work together.
+
+#### Run the core mocha unit/integration test suite.
 `npm run test`
 
 #### Watch the core mocha unit/integration test suite.
 `npm run test:watch`
 
-#### Run the mocha test suite using the karmer runner.
+### DOM API level
+
+Testing the components at the React level isn't enough.
+We need to make sure they will behave as expected with a **real DOM**.
+To solve that problem we use [karma](https://github.com/karma-runner/karma).
+It's almost a drop in replacement of [jsdom](https://github.com/tmpvar/jsdom).
+
+#### Run the mocha test suite using the karma runner.
 `npm run test:karma`
 
-#### Run the e2e selenium tests.
+### Browser API level
+
+In the end, components are going to be used in a real browser.
+The DOM is just one dimension of that environment.
+We need to take into account the rendering engine.
+We are using a [docker](https://github.com/docker/docker) to take screenshots and comparing them with the baseline.
+
+### Run the e2e selenium tests.
 `npm run test:e2e`
 
-#### Run the visual regression tests
+### Run the visual regression tests
 `npm run test:regressions`
 
```

Comment:
For completeness a quick description of the difference between the e2e and regression tests would be good.